core/src/parameter.jl

```diff
@@ -116,16 +116,20 @@ end
 """
 Type for storing metadata of edges in the graph:
 id: ID of the edge (only used for labeling flow output)
+flow_idx: Index in the vector of flows
 type: type of the edge
 subnetwork_id_source: ID of subnetwork where this edge is a source
   (0 if not a source)
 edge: (from node ID, to node ID)
+basin_idxs: Basin indices of source and destination nodes (0 if not a basin)
 """
 struct EdgeMetadata
     id::Int32
+    flow_idx::Int32
     type::EdgeType.T
     subnetwork_id_source::Int32
     edge::Tuple{NodeID, NodeID}
+    basin_idxs::Tuple{Int32, Int32}
```

Comment:
I would advise to split this into basin_source_idx and basin_destination_idx attributes. Code later on makes repeated use of [1] or [2] which is less readable (especially when it's unclear that something returns EdgeMetadata). Or you could make it into a namedtuple?